src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -226,93 +171,117 @@ export default function TimeSeriesChart(props: TimeSeriesChartProps) {
                     }
-                    {!dataToDisplay &&
-                        <LoadingIndicator />
+                    <CartesianGrid vertical={props.chartType !== "Bar"} strokeDasharray="2 4" />
+                    <YAxis
+                        tickFormatter={tickFormatter}
+                        axisLine={false}
+                        interval={0}
+                        tickLine={false}
+                        width={32}
+                        domain={yAxisDomain}
+                    />
+                    <XAxis
+                        id="myXAxis"
+                        domain={[xAxisTicks![0], xAxisTicks![xAxisTicks!.length - 1]]}
+                        padding={props.chartType === 'Bar' ? 'gap' : { left: 0, right: 0 }}
+                        tick={DayTick}
+                        scale={'time'}
+                        type={'number'}
+                        axisLine={false}
+                        dataKey="timestamp"
+                        tickMargin={0}
+                        minTickGap={0}
+                        tickLine={false}
+                        ticks={xAxisTicks}
+                        includeHidden
+                        interval={0}
+                    />
+                    {props.children}
+                    {props.chartHasData &&
+                        <>
+                            {props.chartType === "Line" &&
+                                <>
+                                    {createLineChartDefs(layoutContext, gradientKey, props.series, props.options)}
+                                    {(props.options as MultiSeriesLineChartOptions)?.thresholds?.filter(t => t.referenceLineColor)?.map(threshold =>
+                                        <ReferenceLine y={threshold.value} stroke={resolveColor(layoutContext.colorScheme, threshold.referenceLineColor)} />
+                                    )}
+                                    {keys.map((dk, i) =>
+                                        <Line
+                                            connectNulls={(props.options as MultiSeriesLineChartOptions)?.connectNulls}
+                                            strokeWidth={2}
+                                            key={`${gradientKey}${i}`}
+                                            type="monotone" dataKey={dk}
+                                            dot={!((props.options as MultiSeriesLineChartOptions)?.hideDots ?? false)}
+                                            stroke={`url(#${gradientKey}${i})`}
+                                        />
+                                    )}
+                                </>
+                            }
+                            {props.chartType === "Bar" &&
+                                <>
+                                    {createBarChartDefs(layoutContext, gradientKey, props.series, props.options)}
```

Comment:
The bar and area defs creation was moved to helper functions (like line defs) to keep this code a little smaller/cleaner looking.